### DISCUSSION TITLE
problem on redirect_uri while deploy lowdefy app with keycloak on https service

### DISCUSSION THREAD

**Discussion by 13316090972** [lowdefy/lowdefy#1213] [Questions and Help]
I am trying to deploy a lowdefy app with keycloak on https service. So I add a nginx to reverse proxy lowdefy app from http 8848 port to https 9500 port, and config the https 9500 port on "Valid Redirect URIs" in the keycloak .
![image](https://user-images.githubusercontent.com/56707831/170911411-e425b475-3954-4e7a-a726-70e5fb230890.png)
![1653880579(1)](https://user-images.githubusercontent.com/56707831/170911861-065854ce-6d17-4410-99f3-ff809c776cf0.jpg)

1. I visit 9500 port with htts protocal.
![1653880736(1)](https://user-images.githubusercontent.com/56707831/170911878-3e69ee9e-ba76-4eb0-b86e-f29bd74cd0a8.jpg)

2. The browser jump to keycloak login page,9080 port.
![1653880736(1)](https://user-images.githubusercontent.com/56707831/170911920-7940c7a9-812a-4f97-a4d5-ad2fdfa16e38.jpg)

4. I input the use and password .
5. The browser occur error:The plain HTTP request was sent to HTTPS port
![1653880697(1)](https://user-images.githubusercontent.com/56707831/170911927-80bd70f8-37cd-41cf-97fe-e9aa339afc3f.jpg)

**Comment by SamTolmay** [comment]
Hi @13316090972 

Lowdefy is supposed to use http for the redirect_uri when running the dev server (when running `lowdefy dev` or `npx lowdefy@latest dev`), and use https for production servers, like the node or docker servers. Are you running a development or production server? You should not be running a development server in production.

If you are experiencing the issue while developing, maybe this stackoverflow issue on Nginx configuration could help:
https://stackoverflow.com/questions/8768946/dealing-with-nginx-400-the-plain-http-request-was-sent-to-https-port-error

In these cases port 80 was configured to use ssl.
